React/Base/RCTConvert.m

```diff
@@ -1043,7 +1039,28 @@ + (RCTUIColor *)UIColor:(id)json // TODO(macOS GH#750)
       RCTUIColor *highContrastDarkColor = [RCTConvert UIColor:highContrastDark]; // TODO(macOS GH#750)
       if (lightColor != nil && darkColor != nil) {
 #if TARGET_OS_OSX
-        RCTDynamicColor *color = [[RCTDynamicColor alloc] initWithAquaColor:lightColor darkAquaColor:darkColor];
+				NSColor *color = [NSColor colorWithName:nil dynamicProvider:^NSColor * _Nonnull(NSAppearance * _Nonnull appearance) {
+					NSMutableArray<NSAppearanceName> *appearances = [NSMutableArray arrayWithArray:@[NSAppearanceNameAqua,NSAppearanceNameDarkAqua]];
+					if (highContrastLightColor != nil) {
+						[appearances addObject:NSAppearanceNameAccessibilityHighContrastAqua];
+					}
+					if (highContrastDarkColor != nil) {
+						[appearances addObject:NSAppearanceNameAccessibilityHighContrastDarkAqua];
+					}
+					NSAppearanceName bestMatchingAppearance = [appearance bestMatchFromAppearancesWithNames:appearances];
+					if (bestMatchingAppearance == NSAppearanceNameAqua) {
+						return lightColor;
+					} else if (bestMatchingAppearance == NSAppearanceNameDarkAqua) {
+						return darkColor;
+					} else if (bestMatchingAppearance == NSAppearanceNameAccessibilityHighContrastAqua) {
+						return highContrastLightColor;
+					} else if (bestMatchingAppearance == NSAppearanceNameAccessibilityHighContrastDarkAqua) {
+						return highContrastDarkColor;
+					} else {
+						RCTLogConvertError(json, @"a UIColor. Expected a dynamic appearance aware color.");
```

Comment:
should it be NSColor instead of UIColor?

Reply:
This log statement is actually misleading, and should never be hit. I think I'll just remove it..